### ISSUE TITLE
Feature: Integrate GPU Node and Deploy Self-Hosted AI Stack (RAG)

### ISSUE THREAD

**Issue by binaryn3xus** [binaryn3xus/HomeOps#3789]
## 🚀 Overview
Expand the HomeOps cluster to support hardware-accelerated AI workloads. This involves bringing the existing Linux machine with the **NVIDIA P2000** into the cluster as a dedicated worker node and deploying a Retrieval-Augmented Generation (RAG) stack to leverage the `ProductDescription` dataset.

## 🎯 Goals
- [ ] **Infrastructure:** Join the P2000 machine to the Talos cluster with NVIDIA extensions.
- [ ] **Orchestration:** Configure NVIDIA device plugins for Kubernetes resource scheduling.
- [ ] **Inference:** Deploy a self-hosted LLM engine (Ollama) optimized for CUDA.
- [ ] **Data:** Deploy a Vector Database (Qdrant) to handle `embeddings` and `chunks` for the product data.

## 🏗️ Proposed Architecture
We will follow a RAG (Retrieval-Augmented Generation) pattern. The P2000 will handle the heavy lifting of inference and embedding generation, while Flux manages the lifecycle of the AI components.

---

### Phase 1: Hardware & Node Prep (Talos)
- [ ] Generate a new Talos machine configuration using the `nonfree-kmod-nvidia` and `nvidia-container-toolkit` extensions.
- [ ] Provision the P2000 machine and join it to the cluster.
- [ ] Label the node: `kubectl label node <node-name> hardware-type=gpu`.
- [ ] (Optional) Taint the node to prevent non-AI workloads from consuming GPU-node resources: `node-role.kubernetes.io/gpu:NoSchedule`.

### Phase 2: GPU Operator & Plugins
- [ ] Add the `nvidia-device-plugin` HelmRelease to `kubernetes/apps`.
- [ ] Verify GPU availability:
  ```bash
  kubectl describe node <gpu-node-name> | grep [nvidia.com/gpu](https://nvidia.com/gpu)

### Phase 3: AI Stack Deployment (via Flux)
- ​[ ] Inference Engine: Deploy Ollama via Helm. Ensure resource limits request nvidia.com/gpu: 1.
- ​[ ] Vector Database: Deploy Qdrant in the apps namespace.
- ​[ ] Frontend: Deploy Open WebUI for an internal interface to interact with the models.

### ​Phase 4: Data Integration (ProductDescription)
-​ [ ] Create a workflow to ingest data from the ProductDescription table.
-​ [ ] Perform chunking and generate embeddings using a local model (e.g., nomic-embed-text).
- ​[ ] Upsert the embeddings and chunks into the Qdrant collection.

​📚 References
​Talos Linux NVIDIA Guide
​Ollama Helm Chart
​NVIDIA Device Plugin

​Notes: The P2000 provides 5GB of dedicated VRAM, which is significantly better for this task than the shared-memory Iris Xe iGPUs on the MS-01 nodes.

**Comment by nicolapierre** [comment]
Thank you for your suggestion to integrate a GPU node and deploy a self-hosted AI stack with Retrieval-Augmented Generation (RAG).

To proceed with this feature request, please consider the following steps:

1. **Infrastructure Setup**:
   - Ensure your infrastructure supports GPU nodes. This could be on-premises hardware or cloud providers like AWS, GCP, or Azure, which offer GPU instances.
   - Install necessary drivers and CUDA libraries required for the GPU to function efficiently.

2. **Environment Configuration**:
   - Use containerization tools like Docker to maintain a consistent environment across deployments. Create a Dockerfile that includes all dependencies, such as CUDA and relevant AI libraries (e.g., PyTorch, TensorFlow).
   - Consider Kubernetes for orchestrating your deployments if you expect to scale across multiple nodes.

3. **RAG Model Deployment**:
   - Select an appropriate RAG model that suits your use case. You may consider using pre-trained models from Hugging Face's Transformers library or other similar repositories.
   - Fine-tune your model if necessary to improve performance on your specific dataset.

4. **Integration with GPU**:
   - Ensure that your AI model is set to utilize the GPU. In frameworks like PyTorch, this involves sending your model and data to the GPU using `.to('cuda')`.
   - Test the model's performance on the GPU to confirm that it is significantly faster compared to CPU execution.

5. **Testing and Validation**:
   - Test your deployment in a controlled environment to ensure all components work together seamlessly.
   - Validate the inference results to ensure they meet your accuracy and performance criteria.

6. **Monitoring and Maintenance**:
   - Implement monitoring tools to keep track of GPU utilization and system performance.
   - Regularly update your models and dependencies to incorporate improvements and security patches.

Feel free to reach out if you need further assistance during any of these steps. Your contribution to enhancing the project's capabilities is greatly appreciated.

**Comment by binaryn3xus** [comment]
Complete
